### PR TITLE
Refactor instructor assessment groups page

### PR DESCRIPTION
**Pull request by nwalters512**
Prepping for some improvements to this page, namely, actually doing POST-Redirect-GET instead of the funky respond-directly-to-POST thing this page does at the moment. This will rely on the flash messages being introduced in #7905 instead of rendering error messages directly in the response.

### CODE REVIEW

**Comment by jonatanschroeder** [comment]
Given #6256, do you want to at least remove the copy button to avoid confusion, assuming you don't want to fix it in this PR?

**Review by jonatanschroeder — APPROVED** [review]
I'm approving as from my tests it seems to be working as it did before, but I'm assuming you'll handle the POST-redirect-to-GET in a separate PR.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js [hunk 1]
```diff
@@ -25,144 +29,156 @@ function obtainInfo(req, res, next) {
...
+        res.locals.assessment.id,
+        res.locals.authn_user.user_id,
+      ]);
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'add_group') {
+      const assessment_id = res.locals.assessment.id;
+      const group_name = req.body.group_name;
+      if (!group_name || String(group_name).length < 1) {
+        res.locals.errormsg = 'Please enter a group name when adding a group';
+        await obtainInfo(req, res);
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js:144]: Yes, I'm going to do proper POST/redirect/GET in a separate PR. This one is cleaning things up so it's slightly easier to make that change. I'm going to remove the "copy" button in another PR as well.

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js [hunk 2]
```diff
@@ -25,144 +29,156 @@ function obtainInfo(req, res, next) {
...
-      });
-    });
-  });
+    },
+    z.object({
+      group_id: IdSchema,
+      name: z.string(),
+      size: z.number(),
+      // TODO: what actually is the type of this thing?
+      uid_list: z.array(z.string()),
```
Comment (mwest1066) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js:69]: That seems like the correct type to me.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js:69]: Thanks for the reminder to remove this - left over from a WIP!

File: apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js [hunk 3]
```diff
@@ -25,144 +29,156 @@ function obtainInfo(req, res, next) {
...
-        });
-      });
-    });
-  });
+    },
+    z.object({
+      group_id: IdSchema,
+      name: z.string(),
+      size: z.number(),
+      // TODO: what actually is the type of this thing?
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js:68]:
```suggestion
```

**Pull request merged**